diff --git a/extension.js b/extension.js
index a165951..f222338 100644
--- a/extension.js
+++ b/extension.js
@@ -336,7 +336,7 @@ class MprisLabel extends PanelMenu.Button {
 		}
 
 		const icon = Gio.Icon.new_for_string(this._setVolumeIcon(volumeRatio));
-		Main.osdWindowManager.show(monitor, icon, streamName, volumeRatio);
+		Main.osdWindowManager.showAll(monitor, icon, streamName, volumeRatio);
 	}
 
 	_getStream(){
